js/vcf.js

```diff
@@ -75,21 +75,28 @@ function telephone(type, len) {
     return "TEL;TYPE=" + type.toUpperCase() + ",VOICE:" + rndNum(len) + "\n";
 }
 
-function photo(b64Arr, type, put) {
+function photo(type, put) {
     if(put === true) {
-        var pic = b64Arr[type][Math.floor(Math.random() * b64Arr[type].length)];
+        var pic = photos[type][Math.floor(Math.random() * photos[type].length)];
         return "PHOTO;TYPE=" + type.toUpperCase() + ";ENCODING=B:" + pic + "\n";
     } else {
         return "";
     }
 }
 
-function addresses(addressTypes) {
+function addresses(addressTypes, generateAddressType) {
     var result = "";
     var length = addressTypes.length;
     for(i = 0; i < length; i++) {
         var type = addressTypes[i].toUpperCase();
-        result += `ADR;TYPE=${type}:;;${rndNum(3)} ${rndStrUp(6)} ${rndStrUp(4)};${rndStrUp(8)};${rndStrUp(5)};${rndNum(5)};${rndStrUp(9)}\n`;
+        if(generateAddressType === "random string") {
+            result += `ADR;TYPE=${type}:;;${rndNum(3)} ${rndStrUp(6)} ${rndStrUp(4)};${rndStrUp(8)};${rndStrUp(5)};${rndNum(5)};${rndStrUp(9)}\n`;
+        } else if(generateAddressType === "real addresses") {
+            var address = realAddresses[Math.floor(Math.random() * realAddresses.length)];
```

Comment:
This could be moved to separate function. Selecting random element of an array is done few times in the code.